src/test/libcephfs/test.cc

```diff
@@ -3746,3 +3746,26 @@ TEST(LibCephFS, SnapdirAttrsOnSnapRename) {
   ASSERT_EQ(0, ceph_unmount(cmount));
   ceph_shutdown(cmount);
 }
+
+TEST(LibCephFS, BuffersOverLimit) {
+  struct ceph_mount_info *cmount;
+  ASSERT_EQ(0, ceph_create(&cmount, NULL));
+  ASSERT_EQ(0, ceph_conf_read_file(cmount, NULL));
+  ASSERT_EQ(0, ceph_conf_parse_env(cmount, NULL));
+  ASSERT_EQ(0, ceph_mount(cmount, "/"));
+
+  char test_file[1024];
+  sprintf(test_file, "test_buffers_over_limit_%d", getpid());
+  int fd = ceph_open(cmount, test_file, O_CREAT|O_RDWR, 0666);
+  ASSERT_LT(0, fd);
+
+  const size_t BUFSIZE = static_cast<size_t>(3) * 1024 * 1024 * 1024;
+  auto out_buf = std::make_unique<char[]>(BUFSIZE);
```

Comment:
Does this really need to be a unique_ptr?

Reply:
buffers are large, I cannot allocate such huge memory on stack like it is being don in other test cases where the sizes are tiney like:
```
  char out0[] = "hello ";
  char out1[] = "world\n";
```

For heap, I cannot use `new` and `delete` since they might leak in case of a abrupt failure. `unique_ptr` is safe and release memory when gets out of scope.

Reply:
The abrupt failure is going to crash the process anyway. Are you worried about limit of memory allocation in process stack?

Reply:
> The abrupt failure is going to crash the process anyway. Are you worried about limit of memory allocation in process stack?

yes, that's the only concern, have ran out of stack memory couple of times while testing.